sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/implementation/websocket/WebSocketClientHandler.java

```diff
@@ -46,80 +52,136 @@ ChannelFuture handshakeFuture() {
     }
 
     @Override
-    public void handlerAdded(ChannelHandlerContext ctx) {
-        handshakeFuture = ctx.newPromise();
+    public void handlerAdded(ChannelHandlerContext context) {
+        handshakeFuture = context.newPromise();
+        compositeByteBuf = context.alloc().compositeBuffer();
     }
 
     @Override
-    public void channelActive(ChannelHandlerContext ctx) {
-        handshaker.handshake(ctx.channel());
+    public void handlerRemoved(ChannelHandlerContext ctx) {
+        try {
+            final BinaryData data = BinaryData.fromListByteBuffer(Arrays.asList(compositeByteBuf.nioBuffers()));
```

Comment:
Should we check that `Arrays.asList(compositeByteBuf.nioBuffers())` is not empty before decoding? For instance, if `processMessage` emits message and releases the buffer, then if Netty calls `handlerRemoved`, it may leave us with empty BinaryData, causing issues with `decode(..)` / or confusing call to message-handler?.